draft-ietf-dprive-dnsoquic.md

```diff
@@ -574,7 +575,8 @@ is defined by two transport parameters:
   a MAX_STREAM_DATA frame.
 
 For better performance, it is RECOMMENDED that clients and servers set each of
-these two parameters to a value of 65535 or greater.
+these two parameters to a value of 65537 or greater, allowing for at least one
+query or response of maximum length 65535 bytes.
```

Comment:
I missed this before, but this assumes that a) you want maximum-sized queries/responses and b) in that case, you only have one concurrent request.  Neither of those seems likely, so this recommendation is not likely to be very good.

Reply:
There is one request per stream, so if the client opens a stream to send a query, it automatically grants the initial-bidir credits for that stream to the server. But yes, there is still the issue of global flow control. And, in the XFR case, there is the issue of the sender having to send the entire zone content in a stream containing multiple responses. Suggestion?

Reply:
The fewer words thing probably captures the best advice I can give: just say that the initial flow control limits for bidirectional streams will affect the size of requests and responses unless you are willing to spend an additional round trip to open things up (which is probably fine for XFR, so optimize for queries).  The connection-level limit and the number of streams will affect the number of requests that can be made at the same time.  Then, let people reach their own conclusions.

I tend to think that most servers are best setting a low limit for the initial stream data limits, optimizing for queries.  XFR can use MAX_STREAM_DATA to open things out: even if it costs a round trip, it will still be faster than TCP.

But let's not pretend that we will know better than the people implementing and deploying this stuff: they will be able to tune to their conditions far better than any generic parameters we might be able to put in an RFC.  Our job is just to give them the right pointers.

Reply:
I get your point: the advice here is partly a guess, and we will know more after actual deployments. I still think we need some text to draw attention to the issues, but I accept that the advice would have to be just that. In my mind, this means the following: 
* Point out that QUIC transport allows multiple levels of flow control, such as how many streams can be created, how much data can be sent on a stream, and how much data can be sent on the union of all streams.
* For DoQ, controlling how many streams are created allow servers to control how many new requests the client can send on a given connection.
* The stream data control allow servers to control how much data can be sent by clients. If that control is strict, clients will have to wait for new credits before they can transmit the full content of the request.
* The stream data control allow clients to control how much data can be sent by servers. If that control is strict, servers will have to wait for new credits before they can transmit the full content of the responses.
* Initial values of parameters control how many requests and how much data can be sent by clients and servers at the beginning of the connection. The parameter values received in the initial connection also control how many requests and how much data can be sent by clients using 0-RTT data in a resumed connection.
And then, add text explaining that for smooth operation it might be good to set initial values of parameters (give the list) that allow the client to send at least some requests, and the server to send the responses without being blocked. May not need to provide numerical values. May need to point out the slow loris attack.

Reply:
Please check the new text.